cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -410,82 +410,91 @@ public StreamConfig create(final StreamId streamId, @Nullable final Properties p
     // User should have write access to the namespace
     NamespaceId streamNamespace = streamId.getParent();
     ensureAccess(streamNamespace, Action.WRITE);
-    // revoke privileges to make sure there is no orphaned privileges
-    privilegesManager.revoke(streamId);
+
     final Properties properties = (props == null) ? new Properties() : props;
+    String ownerPrincipal = properties.containsKey(Constants.Security.PRINCIPAL) ?
+      properties.getProperty(Constants.Security.PRINCIPAL) : null;
+
+    if (exists(streamId)) {
+      // if stream exists then make sure owner for this create request is same
+      verifyOwner(streamId, ownerPrincipal);
+      // stream create is an idempotent operation as of now so just return null and don't do anything
+      return null;
+    }
+    // revoke privileges to make sure there is no orphaned privileges as the stream doesn't exist its safe to do so
+    // also safe to call here as even if the revoke throws exception we haven't done any changes yet
+    privilegesManager.revoke(streamId);
```

Comment:
It will not fail if stream doesn't exists from our extension although there's  nothing stopping a user who is using a custom extension to fail it. I guess we should handle this call to then.